akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/EventStreamClient.scala

```diff
@@ -105,17 +105,19 @@ object EventStreamClient {
   )(implicit ec: ExecutionContext, mat: Materializer): Source[A, NotUsed] = {
 
     def getAndHandleEvents = {
-      def getAndHandle(lastEventId: Option[String]) = {
+      def getAndHandle(lastEventId: Option[String])
+        : (Future[Option[ServerSentEvent]], A) = {
         import EventStreamUnmarshalling._
         val request = {
           val r = Get(uri).addHeader(Accept(`text/event-stream`))
           lastEventId.foldLeft(r)((r, i) => r.addHeader(`Last-Event-ID`(i)))
         }
-        val events = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val events      = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val emptySource = Future.successful(Source.empty[ServerSentEvent])
         Source
-          .fromFuture(events)
+          .fromFuture(events.fallbackTo(emptySource))
```

Comment:
I didn't know `Sink.lastOption` when I wrote `LastElement`. Maybe it didn't exist back then, because noone on the Akka team stepped up and told me when I added it to akka-stream-contrib. I totally agree with you: using `lastOption` in combination with some recovery mechanism is the way to go. Thanks a lot, mate!